lang/src/jmh/kotlin/org/partiql/lang/PartiQLBenchmark.kt

```diff
@@ -0,0 +1,163 @@
+package org.partiql.lang
+
+import com.amazon.ion.system.IonSystemBuilder
+import org.openjdk.jmh.annotations.Benchmark
+import org.openjdk.jmh.annotations.BenchmarkMode
+import org.openjdk.jmh.annotations.Level
+import org.openjdk.jmh.annotations.Mode
+import org.openjdk.jmh.annotations.OutputTimeUnit
+import org.openjdk.jmh.annotations.Scope
+import org.openjdk.jmh.annotations.Setup
+import org.openjdk.jmh.annotations.State
+import org.openjdk.jmh.annotations.TearDown
+import org.openjdk.jmh.infra.Blackhole
+import org.partiql.lang.eval.EvaluationSession
+import org.partiql.lang.syntax.SqlParser
+import java.util.concurrent.TimeUnit
+
+
+/**
+ * These are the sample benchmarks to demonstrate how JMH benchmarks in PartiQL should be set up.
+ * Refer this [JMH tutorial](http://tutorials.jenkov.com/java-performance/jmh.html) for more information on [Benchmark]s,
+ * [BenchmarkMode]s, etc.
+ */
+open class PartiQLBenchmark {
+    @State(Scope.Thread)
+    open class MyState {
+        var a = 1
+        var b = 2
+        var sum = 0
+    }
+
+    /**
+     * Use [State] objects to initialize some variables that your benchmark code needs,
+     * but which you do not want to be part of the code your benchmark measures.
+     */
+    @Benchmark
+    fun testMethod(state: MyState) {
+        // This is a demo/sample template for building your JMH benchmarks. Edit as needed.
+        // Put your benchmark code here.
+        state.sum = state.a + state.b
+    }
+
+    /**
+     * State class with [Setup] and [TearDown] annotations.
+     *
+     * The [Setup] annotation tell JMH that this method should be called to setup the state object
+     * before it is passed to the benchmark method.
+     * The [TearDown] annotation tells JMH that this method should be called to
+     * clean up ("tear down") the state object after the benchmark has been executed.
+     *
+     * The setup and tear down execution time is not included in the benchmark runtime measurements.
+     */
+    @State(Scope.Thread)
+    open class MyState2 {
+        @Setup(Level.Trial)
+        fun doSetup() {
+            sum = 0
+            println("Do Setup")
+        }
+
+        @TearDown(Level.Trial)
+        fun doTearDown() {
+            println("Do TearDown")
+        }
+
+        var a = 1
+        var b = 2
+        var sum = 0
```

Comment:
I'm sure this was helpful for getting it working on your machine but it doesn't belong in `main`.